js/packages/teams-ai/src/models/OpenAIModel.ts

```diff
@@ -374,10 +410,73 @@ export class OpenAIModel implements PromptCompletionModel {
             this._events.emit('responseReceived', context, memory, response);
             return response;
         } catch (err: unknown) {
+            console.log(err);
             return this.returnError(err, input);
         }
     }
 
+    /**
+     * Converts the messages to ChatCompletionMessageParam[].
+     * @param {Message<string>} messages - The messages from result.output.
+     * @returns {ChatCompletionMessageParam[]} - The converted messages.
+     */
+    private convertMessages(messages: Message<string>[]): ChatCompletionMessageParam[] {
+        const params: ChatCompletionMessageParam[] = [];
+        // Iterate through the messages and check for action calls
+
+        for (const message of messages) {
+            let param: ChatCompletionMessageParam = {
+                role: 'user',
+                content: ''
+            };
+
+            if (message.role === 'user') {
+                param.content = message.content ?? '';
+            } else if (message.role === 'system') {
+                param = {
+                    role: 'system',
+                    content: message.content ?? ''
+                };
+            } else if (message.role === 'assistant') {
+                param = {
+                    role: 'assistant',
+                    content: message.content ?? ''
+                };
+                const toolCallParams: ChatCompletionMessageToolCall[] = [];
+
+                if (message.action_calls && message.action_calls.length > 0) {
+                    for (const toolCall of message.action_calls) {
+                        toolCallParams.push({
+                            id: toolCall.id,
+                            function: {
+                                name: toolCall.function.name,
+                                arguments: toolCall.function.arguments
+                            },
+                            type: toolCall.type
+                        });
+                    }
+
+                    param.tool_calls = toolCallParams;
+                }
+            } else if (message.role === 'tool') {
+                param = {
+                    role: 'tool',
+                    content: message.content ?? '',
+                    tool_call_id: message.action_call_id ?? ''
+                };
+            } else {
+                param = {
+                    role: 'function',
```

Comment:
don't think we use functions anymore

Reply:
Added work item: https://github.com/microsoft/teams-ai/issues/1946